### PR TITLE
[FIX] display the right logo in e-mail using multi-company

### PR DESCRIPTION
**Pull request by mart-e**
sending an e-mail with a user that own more than one company didn't place the right logo in the header of the mail.

opw-772405

### CODE REVIEW

File: addons/account/data/invoice_action_data.xml [hunk 1]
```diff
@@ -62,8 +62,7 @@ from ${object.company_id.name}.
             <field name="auto_delete" eval="True"/>
             <field name="body_html"><![CDATA[<html>
                 <head></head>
-                % set record = ctx.get('record')
-                % set company = record and record.company_id or user.company_id
+                % set company = ctx.get('company') or user.company_id
```
Comment (nim-odoo) [inline addons/account/data/invoice_action_data.xml]: should be `user.company_id.id` to be consistent

File: addons/sale/data/mail_template_data.xml [hunk 2]
```diff
@@ -63,8 +63,7 @@ from ${object.company_id.name}.
             <field name="auto_delete" eval="True"/>
             <field name="body_html"><![CDATA[<html>
                 <head></head>
-                % set record = ctx.get('record')
-                % set company = record and record.company_id or user.company_id
+                % set company = ctx.get('company') or user.company_id
```
Comment (nim-odoo) [inline addons/sale/data/mail_template_data.xml]: same as above: should be `user.company_id.id` to be consistent

File: addons/mail/models/res_partner.py [hunk 3]
```diff
@@ -81,9 +80,16 @@ def _notify_prepare_template_context(self, message):
         if message.res_id and message.model in self.env:
             record = self.env[message.model].browse(message.res_id)
 
+        if record and hasattr(record, 'company_id'):
+            company_id = record.company_id.id
+        else:
+            company_id = user.company_id.id
+        company_name = user.company_id.name
```
Comment (nim-odoo) [inline addons/mail/models/res_partner.py]: Hum, there might be an inconsistency between company_id and company_name, then.
Could be:
        if record and hasattr(record, 'company_id'):
            company = record.company_id
        else:
            company = user.company_id
        company_id = company.id
        company_name = company.name

File: addons/account/data/invoice_action_data.xml (hunk 1 above)
Comment (mart-e) [reply addons/account/data/invoice_action_data.xml]: well I was wondering which was best. It implies changing also the `/logo.png?company=${company.id}` below

**Pull request closed without merging**